client/src/components/authentification/Register.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import { Redirect, Link } from "react-router-dom";
+import { RootStateOrAny, useSelector, useDispatch } from "react-redux";
+
+import { makeStyles } from "@material-ui/core/styles";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+
+import { register } from "../../actions/actions";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    "& > *": {
+      margin: theme.spacing(1),
+      width: "25ch",
+    },
+  },
+}));
+
+const Register: React.FC = () => {
+  const dispatch = useDispatch();
+  const classes = useStyles();
+
+  const isAuthenticated = useSelector(
+    (state: RootStateOrAny) => state.auth.isAuthenticated
+  );
```

Comment:
You can create `selectors` folder, where you can write all selectors functions

Reply:
done